### PR TITLE
[AMS-1619] lightweight API methods for objects

### PR DESCRIPTION
**Pull request by xakep139**

### CODE REVIEW

File: src/VStore.Host/Controllers/ObjectsController.cs [hunk 1]
```diff
@@ -121,6 +121,36 @@ public sealed class ObjectsController : VStoreController
...
         }
 
+        /// <summary>
+        /// Get object's latest version metadata
+        /// </summary>
+        /// <param name="id">Object identifier</param>
+        /// <param name="ifNoneMatch">Object version to check if it has been modified (optional)</param>
+        /// <returns>No body with status 200 Ok or 404 Not Found or 304 Not Modified</returns>
+        [HttpGet("{id:long}")]
+        [ResponseCache(Duration = 120)]
```
Comment (xakep139) [inline src/VStore.Host/Controllers/ObjectsController.cs]: I don't think it's a good idea to set cache headers in methods without specific `versionId` (`GetMetadata()` and `Get()`)

File: src/VStore.Host/Controllers/ObjectsController.cs [hunk 2]
```diff
@@ -207,10 +261,6 @@ public sealed class ObjectsController : VStoreController
             {
                 return NotFound();
             }
-            catch (ObjectInconsistentException ex)
```
Comment (xakep139) [inline src/VStore.Host/Controllers/ObjectsController.cs]: We couldn't get this exception

File: src/VStore/Objects/ObjectsStorageReader.cs [hunk 3]
```diff
@@ -206,11 +204,21 @@ async Task<(bool IsTruncated, int NextVersionIndex, string NextVersionIdMarker)>
             return records;
         }
 
-        public async Task<IReadOnlyCollection<VersionedObjectDescriptor<string>>> GetObjectLatestVersions(long id)
```
Comment (xakep139) [inline src/VStore/Objects/ObjectsStorageReader.cs]: I separated this method into two methods - one for object and one for elements

File: src/VStore/S3/KeyTypesExtensions.cs [hunk 4]
```diff
@@ -28,14 +27,6 @@ public static long AsSubObjectId(this string key)
             return long.Parse(key.Substring(separatorIndex + 1));
         }
 
-        public static string AsS3LockKey(this long id) => string.Concat(id.ToString(), LockFinaliter);
```
Comment (xakep139) [inline src/VStore/S3/KeyTypesExtensions.cs]: Unused since we migrate to RedLock

**Comment by xakep139** [comment]
@denisivan0v should we add new API version for created methods?

File: src/VStore.Host/Controllers/ObjectsController.cs (hunk 1 above)
Comment (denisivan0v) [reply src/VStore.Host/Controllers/ObjectsController.cs]: Agreed

File: src/VStore.Host/Controllers/ObjectsController.cs [hunk 5]
```diff
@@ -121,6 +121,36 @@ public sealed class ObjectsController : VStoreController
             }
         }
 
+        /// <summary>
+        /// Get object's latest version metadata
+        /// </summary>
+        /// <param name="id">Object identifier</param>
+        /// <param name="ifNoneMatch">Object version to check if it has been modified (optional)</param>
+        /// <returns>No body with status 200 Ok or 404 Not Found or 304 Not Modified</returns>
+        [HttpGet("{id:long}")]
```
Comment (denisivan0v) [inline src/VStore.Host/Controllers/ObjectsController.cs]: Did you mean `HEAD`?

Comment (xakep139) [reply src/VStore.Host/Controllers/ObjectsController.cs]: done in https://github.com/2gis/nuclear-vstore/pull/167/commits/e6b76784646622e80d68d542dfa577389f1ce7ef

**Comment by denisivan0v** [comment]
@xakep139 please update this PR from upstream

**Comment by xakep139** [comment]
> @xakep139 please update this PR from upstream

@denisivan0v, done in https://github.com/2gis/nuclear-vstore/pull/167/commits/973f39ad88edf9955945b52c6d904b0b39533f4d

**Pull request merged**